service/src/main/java/com/ge/predix/acs/privilege/management/dao/TitanMigrationManager.java

```diff
@@ -0,0 +1,96 @@
+package com.ge.predix.acs.privilege.management.dao;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+@Component
+@Profile("titan")
+public final class TitanMigrationManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TitanMigrationManager.class);
+
+    @Autowired
+    @Qualifier("resourceRepository")
+    private ResourceRepository resourceRepository;
+
+    @Autowired
+    @Qualifier("resourceHierarchicalRepository")
+    private GraphResourceRepository resourceHierarchicalRepository;
+
+    @Autowired
+    @Qualifier("subjectRepository")
+    private SubjectRepository subjectRepository;
+
+    @Autowired
+    @Qualifier("subjectHierarchicalRepository")
+    private GraphSubjectRepository subjectHierarchicalRepository;
+
+    private static final int PAGE_SIZE = 100;
+
+    public TitanMigrationManager(final ResourceRepository resourceRepository,
+            final GraphResourceRepository resourceHierarchicalRepository, final SubjectRepository subjectRepository,
+            final GraphSubjectRepository subjectHierarchicalRepository) {
+        this.resourceRepository = resourceRepository;
+        this.resourceHierarchicalRepository = resourceHierarchicalRepository;
+        this.subjectRepository = subjectRepository;
+        this.subjectHierarchicalRepository = subjectHierarchicalRepository;
+    }
+
+    public TitanMigrationManager() {
+
+    }
+
+    @PostConstruct
+    public void doMigration() {
+        try {
+
+            // This version vertex is common to both subject and resource repositories. So this check is sufficient to
+            // trigger migrations in both repos.
+            if (this.resourceHierarchicalRepository.getVersion() == 0) {
+                LOGGER.info("Starting attribute migration process to Titan.");
+
+                int totalResources = 0;
+                int totalSubjects = 0;
+                Pageable pageRequest = new PageRequest(0, PAGE_SIZE);
+                do {
+                    Page<ResourceEntity> pageOfResources = resourceRepository.findAll(pageRequest);
+                    List<ResourceEntity> listOfResources = pageOfResources.getContent();
+                    totalResources += pageOfResources.getNumberOfElements();
+                    listOfResources.forEach(item -> item.setId((long) 0));
+                    resourceHierarchicalRepository.save(listOfResources);
```

Comment:
add another log message here, in case there is a exception before all pages are saved.  And keep the log at line 73 which logs the total number.